instantiateStreaming / compileStreaming should accept FetchResponse with FormData

Constellation opened the pull request:
#### e2b279112b8dc0852fd89c5a063f910b17c5cc21
<pre>
instantiateStreaming / compileStreaming should accept FetchResponse with FormData
<a href="https://bugs.webkit.org/show_bug.cgi?id=221248">https://bugs.webkit.org/show_bug.cgi?id=221248</a>
rdar://problem/74132453

Reviewed by Darin Adler.

Load FormData through ReadableStream in WebAssembly.{instantiateStreaming,compileStreaming}.

* LayoutTests/imported/w3c/web-platform-tests/wasm/wasm_stream_compile_test-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/wasm/wasm_stream_instantiate_test-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/wasm/webapi/wasm_stream_compile_test-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/wasm/webapi/wasm_stream_instantiate_test-expected.txt:
* Source/WebCore/bindings/js/JSDOMGlobalObject.cpp:
(WebCore::handleResponseOnStreamingAction):

Canonical link: <a href="https://commits.webkit.org/253395@main">https://commits.webkit.org/253395@main</a>
</pre>

darinadler commented on Source/WebCore/bindings/js/JSDOMGlobalObject.cpp:
```diff
@@ -475,14 +475,7 @@ static JSC::JSPromise* handleResponseOnStreamingAction(JSC::JSGlobalObject* glob
-        // FIXME: Support FormData loading.
-        // https://bugs.webkit.org/show_bug.cgi?id=221248
+    WTF::switchOn(body, [&](Ref<FormData>&) {
         compiler->fail(globalObject, createDOMException(*globalObject, Exception { NotSupportedError, "Not implemented"_s  }));
```
Is “not implemented” still the correct exception here? I am a little puzzled about removing the comment that says this needs to be implemented, but still keeping a “not implemented” exception.

Constellation replied:
We can make it RELEASE_ASSERT_NOT_REACHED.

Constellation replied:
And changed ;)

Constellation committed:
instantiateStreaming / compileStreaming should accept FetchResponse with FormData
https://bugs.webkit.org/show_bug.cgi?id=221248
rdar://problem/74132453

Reviewed by Darin Adler.

Load FormData through ReadableStream in WebAssembly.{instantiateStreaming,compileStreaming}.

* LayoutTests/imported/w3c/web-platform-tests/wasm/wasm_stream_compile_test-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/wasm/wasm_stream_instantiate_test-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/wasm/webapi/wasm_stream_compile_test-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/wasm/webapi/wasm_stream_instantiate_test-expected.txt:
* Source/WebCore/bindings/js/JSDOMGlobalObject.cpp:
(WebCore::handleResponseOnStreamingAction):

Canonical link: https://commits.webkit.org/253395@main
